### PR TITLE
Remove EncodeToStream(..., []unversioned.GroupVersion)

### PR DESCRIPTION
**Pull request by smarterclayton**
Was not being used. Is a signature change and is necessary for post 1.3 work on Templates and other objects that nest objects.

Extracted from #26044

### CODE REVIEW

**Comment by smarterclayton** [comment]
Proposing for 1.3 as a refactor to enable a larger post 1.3 refactor to not disrupt cherry-picking.

File: pkg/runtime/interfaces.go [hunk 1]
```diff
@@ -31,10 +31,9 @@ const (
 )
 
 type Encoder interface {
-	// EncodeToStream writes an object to a stream. Override versions may be provided for each group
-	// that enforce a certain versioning. Implementations may return errors if the versions are incompatible,
-	// or if no conversion is defined.
-	EncodeToStream(obj Object, stream io.Writer, overrides ...unversioned.GroupVersion) error
+	// EncodeToStream writes an object to a stream. Implementations may return errors if the versions are
```
Comment (lavalamp) [inline pkg/runtime/interfaces.go]: Comment doesn't match :)

File: pkg/runtime/serializer/versioning/versioning.go [hunk 2]
```diff
@@ -224,48 +224,31 @@ func (c *codec) Decode(data []byte, defaultGVK *unversioned.GroupVersionKind, in
 
 // EncodeToStream ensures the provided object is output in the right scheme. If overrides are specified, when
```
Comment (lavalamp) [inline pkg/runtime/serializer/versioning/versioning.go]: update comment

**Comment by lavalamp** [comment]
looks safe & less confusing. LGTM

**Comment by smarterclayton** [comment]
Fixed a test compile error from another PR

**Comment by smarterclayton** [comment]
@k8s-bot test this issue #27205

**Comment by smarterclayton** [comment]
@k8s-bot test this issue #27205

**Comment by smarterclayton** [comment]
@k8s-bit test this issue #IGNORE

**Comment by smarterclayton** [comment]
@k8s-bot test this issue #27685

**Comment by goltermann** [comment]
@k8s-bot node e2e test this

**Pull request merged**